nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_depth.py

```diff
@@ -0,0 +1,362 @@
+                    block_indexes_to_remove.append(block_index)
+                    break
+                valid_block_indexes = [group[group_index]]
+                for i in range(group_index + 1, len(group)):
+                    if group[i] in config:
+                        valid_block_indexes.append(group[i])
+                    else:
+                        nncf_logger.debug('The block #{} or #{} did not satisfy requirement of next static block'.
+                                          format(block_index, valid_block_indexes))
+                        for valid_block_index in valid_block_indexes:
+                            block_indexes_to_remove.append(valid_block_index)
+                        break
+            if found:
+                break
+        return block_indexes_to_remove
+
+
+class ElasticDepthMode(Enum):
+    MANUAL = 'manual'
+    AUTO = 'auto'
+
+    @classmethod
+    def from_str(cls, mode: str) -> 'ElasticDepthMode':
+        if mode == ElasticDepthMode.MANUAL.value:
+            return ElasticDepthMode.MANUAL
+        if mode == ElasticDepthMode.AUTO.value:
+            return ElasticDepthMode.AUTO
+        raise RuntimeError(f"Unknown elasticity depth mode: {mode}."
+                           f"List of supported: {[e.value for e in ElasticDepthMode]}")
+
+
+class EDBuilderStateNames:
+    SKIPPED_BLOCKS = 'skipped_blocks'
+    SKIPPED_BLOCKS_DEPENDENCIES = 'skipped_blocks_dependencies'
+    ORDINAL_IDS = 'ordinal_ids'
+    MODE = 'mode'
+
+
+@ELASTICITY_BUILDERS.register(ElasticityDim.DEPTH)
+class ElasticDepthBuilder(SingleElasticityBuilder):
+    _state_names = EDBuilderStateNames
+
+    def __init__(self, elasticity_params: Optional[Dict[str, Any]] = None,
+                 ignored_scopes: Optional[List[str]] = None,
+                 target_scopes: Optional[List[str]] = None):
+        super().__init__(ignored_scopes, target_scopes, elasticity_params)
+        self._mode = ElasticDepthMode.from_str(self._elasticity_params.get('mode', ElasticDepthMode.AUTO.value))
+        self._skipped_blocks = None  # type: Optional[BUILDING_BLOCKS]
+        self._skip_dependencies = None  # type: Optional[GROUPED_BLOCK_IDS]
+        self._ordinal_ids = None  # type: Optional[ORDINAL_IDS]
+        self._max_block_size = self._elasticity_params.get('max_block_size', 50)
+        self._min_block_size = self._elasticity_params.get('min_block_size', 6)
+        self._allow_nested_blocks = self._elasticity_params.get('allow_nested_blocks', False)
+        self._allow_linear_combination = self._elasticity_params.get('allow_linear_combination', False)
+
+        if self._mode == ElasticDepthMode.MANUAL:
+            skipped_blocks = self._elasticity_params.get('skipped_blocks', [])  # type: List[List[str,str]]
+            self._skipped_blocks = [BuildingBlock(*b) for b in skipped_blocks]
+            self._skip_dependencies = get_group_of_dependent_blocks(self._skipped_blocks)
```

Comment:
@asenina @jpablomch skip_dependencies is not needed for Elastic Depth in manual mode anymore.